packages/schematics/angular/utility/project.ts

```diff
@@ -1,48 +0,0 @@
-/**
- * @license
- * Copyright Google Inc. All Rights Reserved.
- *
- * Use of this source code is governed by an MIT-style license that can be
- * found in the LICENSE file at https://angular.io/license
- */
-import { Tree } from '@angular-devkit/schematics';
-import { getWorkspace } from '../utility/config';
-import { ProjectType, WorkspaceProject, WorkspaceSchema } from '../utility/workspace-models';
-
-
-/**
- * Build a default project path for generating.
- * @param project The project to build the path for.
- */
-export function buildDefaultPath(project: WorkspaceProject): string {
-  const root = project.sourceRoot
-    ? `/${project.sourceRoot}/`
-    : `/${project.root}/src/`;
-
-  const projectDirName = project.projectType === ProjectType.Application ? 'app' : 'lib';
-
-  return `${root}${projectDirName}`;
-}
-
-export function getProject<TProjectType extends ProjectType = ProjectType.Application>(
-  workspaceOrHost: WorkspaceSchema | Tree,
-  projectName: string,
-): WorkspaceProject<TProjectType> {
-  const workspace = isWorkspaceSchema(workspaceOrHost)
-    ? workspaceOrHost
-    : getWorkspace(workspaceOrHost);
-
-  return workspace.projects[projectName] as WorkspaceProject<TProjectType>;
-}
-
-// TODO(hans): change this any to unknown when google3 supports TypeScript 3.0.
-// tslint:disable-next-line:no-any
-export function isWorkspaceSchema(workspace: any): workspace is WorkspaceSchema {
-  return !!(workspace && (workspace as WorkspaceSchema).projects);
-}
-
-// TODO(hans): change this any to unknown when google3 supports TypeScript 3.0.
-// tslint:disable-next-line:no-any
-export function isWorkspaceProject(project: any): project is WorkspaceProject {
```

Comment:
Looks like this file is public after all, and used by the material schematics, but it's not documented and it doesn't have a Public API guard.

I'll check if there is any chance to remove the duplicate code while still preserving this public file. 

Should this file be public?

Reply:
I would be double careful when changing these files as many projects providing schematics rely on the utilities. E.g. [ng-bootstrap](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/schematics/ng-add/index.ts#L8).

Reply:
Looks like the material schematics code that uses this file also requires some cleanup:

```
// TODO(jelbourn): figure out if the need for this `as any` is a bug due to two different
// incompatible `WorkspaceProject` classes in @angular-devkit
options.path = buildDefaultPath(project as any);
```

Reply:
I see, well, this calls for a documented Public API Guard in order to avoid the removal of various things without pushing them through a deprecation phase.

Reply:
Looks like we have a story for documenting the Schematics API: https://github.com/angular/angular/issues/29241

Reply:
These are actually private APIs which if unused we can remove.

Reply:
@alan-agius4, looks like the `ng add @angular/localize` schematic is also affected by this PR, but the e2e tests don't cover that package, so I've created a PR to update the schematic to the newer API, please provide feedback: https://github.com/angular/angular/pull/36897